src/ServiceControl.Transports.AzureServiceBusStandard/QueueLengthProvider.cs

```diff
@@ -0,0 +1,114 @@
+            this.managementClient = new ManagementClient(connectionString);
+        }
+
+        public void Process(EndpointInstanceId endpointInstanceId, EndpointMetadataReport metadataReport)
+        {
+            endpointQueueMappings.AddOrUpdate(
+                endpointInstanceId,
+                id => metadataReport.LocalAddress,
+                (id, old) => metadataReport.LocalAddress
+            );
+        }
+
+        public void Process(EndpointInstanceId endpointInstanceId, TaggedLongValueOccurrence metricsReport)
+        {
+        }
+
+        public Task Start()
+        {
+            stop = new CancellationTokenSource();
+
+            pooler = Task.Run(async () =>
+            {
+
+                while (!stop.Token.IsCancellationRequested)
+                {
+                    try
+                    {
+                        Logger.DebugFormat("Querying management client.");
+
+                        var queues = await managementClient.GetQueuesAsync().ConfigureAwait(false);
+                        var lookup = queues.ToLookup(x => x.Path, StringComparer.InvariantCultureIgnoreCase);
+
+                        Logger.DebugFormat("Retrieved details of {0} queues", lookup.Count);
+
+                        await UpdateQueueLengthStore(lookup);
+
+                        Logger.Debug("Waiting for next interval");
+                        await Task.Delay(QueryDelayInterval).ConfigureAwait(false);
+                    }
+                    catch (Exception e)
+                    {
+                        Logger.Error("Error querying Azure Service Bus queue sizes.", e);
+                    }
+                }
+            });
+
+            return TaskEx.Completed;
+        }
+
+        private async Task UpdateQueueLengthStore(ILookup<string, QueueDescription> queueData)
+        {
+            var timestamp = DateTime.UtcNow.Ticks;
+            foreach (var mapping in endpointQueueMappings)
+            {
+                var queue = queueData[mapping.Value].FirstOrDefault();
+                if (queue != null)
+                {
+                    var entries = new[]
+                    {
+                        new RawMessage.Entry
+                        {
+                            DateTicks = timestamp,
+                            Value = (await managementClient.GetQueueRuntimeInfoAsync( queue.Path )).MessageCountDetails.ActiveMessageCount
+                        }
+                    };
+                    queueLengthStore.Store(entries, new EndpointInputQueue(mapping.Key.EndpointName, queue.Path));
+                }
+                else
+                {
+                    Logger.DebugFormat("Endpoint {0} ({1}): no queue length data found for queue {2}", mapping.Key.EndpointName, mapping.Key.InstanceName ?? mapping.Key.InstanceId, mapping.Value);
+                }
+            }
+        }
+
+        public Task Stop()
+        {
+            stop.Cancel();
```

Comment:
`ManagementClient` needs to be closed as well.

Reply:
done